azure-pipelines.yml

```diff
@@ -47,7 +50,9 @@ stages:
           testResultsFiles: '**/test-output-*.xml' 
           searchFolder: '$(System.DefaultWorkingDirectory)'
   - stage: Deploy
-    dependsOn: Test
+    dependsOn: 
+      - Test 
+      - E2E
```

Comment:
E2E depends on Test. We don't need this line.

Reply:
It's not a big deal to explicitly state that it requires both to pass, in fact it makes it more clear.